src/test/java/school/hei/haapi/integration/SecurityIT.java

```diff
@@ -0,0 +1,109 @@
+package school.hei.haapi.integration;
+
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.ContextConfiguration;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import school.hei.haapi.client.HAHttpClient;
+import school.hei.haapi.exception.NotAuthorizedException;
+import school.hei.haapi.integration.conf.CallerData;
+import school.hei.haapi.integration.conf.ContextInitializer;
+import school.hei.haapi.web.model.StudentResource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+@Testcontainers
+@ContextConfiguration(initializers = ContextInitializer.class)
+@AutoConfigureMockMvc
```

Comment:
If you don't use mock anymore, remove this.

However I suggest you continue mocking Cognito responses here. Then separately test cognitoComponent in another unit test where you only check the cognitoComponent::findByEmail.

This way, the authorization tests will always work even if the authentication tests fail (eg if tokens expire)

Reply:
So you suggest me to create an duplicated IT but in one hand using the mocks and in another hand just to get the result ?